lib/lintSource.js

```diff
@@ -66,6 +66,20 @@ module.exports = async function lintSource(stylelint, options = {}) {
 	const config = configForFile.config;
 	const existingPostcssResult = options.existingPostcssResult;
 
+	if (options.cache) {
+		if (!stylelint._fileCache._hashOfConfig) {
+			stylelint._fileCache.calcHashOfConfig(config);
+		}
```

Comment:
[suggest] I feel it a bit smarter to avoid using the private field `_hashOfConfig`. How about checking the field in `calcHashOfConfig()` like this?

```suggestion
		stylelint._fileCache.calcHashOfConfig(config);
```

```js
calcHashOfConfig(config) {
	if (this._hashOfConfig) return; // avoid re-calculation

	// calculate hash...
```

Reply:
Updated `FileCache` that return if has `_hashOfConfig`. [`b9c8a00`](https://github.com/stylelint/stylelint/pull/6356/commits/b9c8a00f7cf4abf58dafd3b0d0660703aa6c913d#diff-83240954c077d79828b7a92f75b9e1c412d2b228b367c5864011f57b937fd26e)